core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -46,11 +46,14 @@ oppia.factory('UrlService', ['$window', function($window) {
     getPathname: function() {
       return this.getCurrentLocation().pathname;
     },
+    // Topic id should be correctly returned from topic editor as well as
+    // story editor, since both have topic id in their url.
     getTopicIdFromUrl: function() {
       var pathname = this.getPathname();
       var pathValues = pathname.split('/');
       if (pathValues.length < 3 ||
-          pathValues[1] !== 'topic_editor') {
+          (pathValues[1] !== 'topic_editor' &&
+          pathValues[1] !== 'story_editor')) {
```

Comment:
This additional check had to be added since topic id can also be requested from a story editor context.

Reply:
The way it's implemented seems a bit ad hoc, though. It would be better to have regexes for topic editor url and story editor url, and match the url to either of these. If there's a match, then do the appropriate processing for that case. If it matches neither, then error.